examples/webgpu_compute.html

```diff
@@ -0,0 +1,208 @@
+
+					const r = Math.random() * 0.01 + 0.0005;
+					const degree = Math.random() * 360;
+					velocityArray[ i * 3 + 0 ] = r * Math.sin( degree * Math.PI / 180 );
+					velocityArray[ i * 3 + 1 ] = r * Math.cos( degree * Math.PI / 180 );
+
+				}
+
+				const particleGroup = new WebGPUBufferGroup( 'particle', new THREE.BufferAttribute( particleArray, 3 ) );;
+				const velocityGroup = new WebGPUBufferGroup( 'velocity', new THREE.BufferAttribute( velocityArray, 3 ) );
+
+				pointer = new THREE.Vector2( - 10.0, - 10.0 ); // Out of bounds first
+
+				const pointerGroup = new WebGPUUniformsGroup( 'mouseUniforms' ).addUniform(
+					new Vector2Uniform( 'pointer', pointer )
+				);
+
+				const computeGroups = [
+					particleGroup,
+					velocityGroup,
+					pointerGroup
+				];
+
+				const computeShader = `#version 450
+					#define PARTICLE_NUM ${particleNum}
+					#define PARTICLE_SIZE ${particleSize}
+					#define ROOM_SIZE 1.0
+					#define POINTER_SIZE 0.1
+
+					// Limitation for now: the order should be the same as groups order
+
+					layout(set = 0, binding = 0) buffer Particle {
+						float particle[ PARTICLE_NUM * PARTICLE_SIZE ];
+					} particle;
+
+					layout(set = 0, binding = 1) buffer Velocity {
+						float velocity[ PARTICLE_NUM * PARTICLE_SIZE ];
+					} velocity;
+
+					layout(set = 0, binding = 2) uniform MouseUniforms {
+						vec2 pointer;
+					} mouseUniforms;
+
+					void main() {
+						uint index = gl_GlobalInvocationID.x;
+						if ( index >= PARTICLE_NUM ) { return; }
+
+						vec3 position = vec3(
+							particle.particle[ index * 3 + 0 ] + velocity.velocity[ index * 3 + 0 ],
+							particle.particle[ index * 3 + 1 ] + velocity.velocity[ index * 3 + 1 ],
+							particle.particle[ index * 3 + 2 ] + velocity.velocity[ index * 3 + 2 ]
+						);
+
+						if ( abs( position.x ) >= ROOM_SIZE ) {
+
+							velocity.velocity[ index * 3 + 0 ] = - velocity.velocity[ index * 3 + 0 ];
+
+						}
+
+						if ( abs( position.y ) >= ROOM_SIZE ) {
+
+							velocity.velocity[ index * 3 + 1 ] = - velocity.velocity[ index * 3 + 1 ];
+
+						}
+
+						if ( abs( position.z ) >= ROOM_SIZE ) {
+
+							velocity.velocity[ index * 3 + 2 ] = - velocity.velocity[ index * 3 + 2 ];
+
+						}
+
+						float dx = mouseUniforms.pointer.x - position.x;
+						float dy = mouseUniforms.pointer.y - position.y;
+						float distanceFromPointer = sqrt( dx * dx + dy * dy );
+
+						if ( distanceFromPointer <= POINTER_SIZE ) {
+
+							position.x = 0.0;
+							position.y = 0.0;
+							position.z = 0.0;
+
+						}
+
+						particle.particle[ index * 3 + 0 ] = position.x;
+						particle.particle[ index * 3 + 1 ] = position.y;
+						particle.particle[ index * 3 + 2 ] = position.z;
+					}
+				`;
+
+				computeParams.push( {
+					num: particleNum,
+					shader: computeShader,
+					groups: computeGroups
+				} );
+
+				const pointsGeometry = new THREE.BufferGeometry().setAttribute(
```

Comment:
How about adding a comment here so it's more clear for users what the compute shader actually does? E.g.
```
Use a compute shader to to animate the point cloud's vertex data.
```

Reply:
Updated, thanks